service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -46,14 +47,42 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		}
 	}
 
-	var clusterCIDR string
-	var subnetID string
-	var networkID string
+	values := map[string]interface{}{
```

Comment:
I moved values up here so we can inject provider-specific values in the switch block below.

Reply:
Kind of hate how much provider specific stuff we still have going on here, but maybe this can be fixed later on.

Reply:
It's a long-time issue. One operator with provider-specific logic or one operator per provider... since we're already doing it I don't feel too bad continuing it here for OpenStack.